addons/l10n_hu_edi/models/account_move.py

```diff
@@ -0,0 +1,1135 @@
+    def _compute_show_reset_to_draft_button(self):
+        super()._compute_show_reset_to_draft_button()
+        self.filtered(lambda m: m.l10n_hu_edi_state not in [False, 'rejected', 'cancelled']).show_reset_to_draft_button = False
+
+    @api.depends('l10n_hu_edi_state')
+    def _compute_need_cancel_request(self):
+        # EXTEND 'account' to add dependencies
+        return super()._compute_need_cancel_request()
+
+    @api.depends('name')
+    def _compute_l10n_hu_edi_attachment_filename(self):
+        for move in self:
+            move.l10n_hu_edi_attachment_filename = f'{move.name.replace("/", "_")}.xml'
+
+    @api.depends('l10n_hu_edi_state')
+    def _compute_l10n_hu_edi_show_button_update_status(self):
+        for move in self:
+            move.l10n_hu_edi_show_button_update_status = move._l10n_hu_edi_get_valid_action() in ['query_status', 'recover_timeout']
+
+    # === Overrides === #
+
+    def action_reverse(self):
+        # EXTEND account
+        unconfirmed = self.filtered(lambda m: m.l10n_hu_edi_state not in ['confirmed', 'confirmed_warning'])
+        if unconfirmed:
+            raise UserError(_(
+                'Invoices %s have not yet been confirmed by NAV. Please wait for confirmation before issuing a modification invoice.',
+                unconfirmed.mapped('name'))
+            )
+        return super().action_reverse()
+
+    def _need_cancel_request(self):
+        # EXTEND account
+        # Technical annulment should be available only in debug mode
+        return super()._need_cancel_request() or (self.l10n_hu_edi_state in ['confirmed', 'confirmed_warning'] and request.session.debug)
+
+    def button_request_cancel(self):
+        # EXTEND 'account'
+        if self._need_cancel_request() and self.l10n_hu_edi_state in ['confirmed', 'confirmed_warning']:
+            return {
+                "name": _("Technical Annulment"),
+                "type": "ir.actions.act_window",
+                "view_type": "form",
+                "view_mode": "form",
+                "res_model": "l10n_hu_edi.cancellation",
+                "target": "new",
+                "context": {"default_invoice_id": self.id},
+            }
+
+        return super().button_request_cancel()
+
+    def _post(self, soft=True):
+        # EXTEND account
+        to_post = self.filtered(lambda move: move.date <= fields.Date.context_today(self)) if soft else self
+        for move in to_post.filtered(lambda m: m.country_code == 'HU' and m.is_sale_document()):
+            move._l10n_hu_edi_set_chain_index_and_line_number()
+        return super()._post(soft=soft)
+
+    # === Actions === #
+
+    def l10n_hu_edi_button_update_status(self, from_cron=False):
+        """ Attempt to update the status of the invoices in `self` """
+        self = self.filtered(lambda m: m._l10n_hu_edi_get_valid_action() in ['recover_timeout', 'query_status'])
+
+        with self._l10n_hu_edi_acquire_lock():
```

Comment:
I probably need to re-think how the locking is done in those methods, and also how many times we commit a transaction. Ideally, we should lock once (in the top-level method) and commit only once as well (after the new state has been written on account.move). 

I agree that locking is not needed if we're just updating the status. I think the only situation where locking is absolutely needed is the upload, because if we can't write the transaction ID to the move it's lost.

Besides that, for some heavy loads (eg. the transaction recovery flow) we might want to proactively lock as well, to avoid losing the work and re-trying the request automatically if a serialization error occurs.